src/common/GasService.sol

```diff
@@ -2,26 +2,132 @@
+        updateShares = BASE_COST + 49968;
+        maxAssetPriceAge = BASE_COST + SMALL_COST;
+        maxSharePriceAge = BASE_COST + SMALL_COST;
     }
 
     /// @inheritdoc IGasService
-    function maxBatchSize(uint16) public view returns (uint128) {
-        return _maxBatchSize;
+    function batchGasLimit(uint16) public view returns (uint128) {
+        return _batchGasLimit;
     }
 
     /// @inheritdoc IGasService
-    function gasLimit(uint16, bytes calldata) public view returns (uint128) {
-        return _messageGasLimit;
+    function messageGasLimit(uint16, bytes calldata message) public view returns (uint128) {
+        MessageType kind = message.messageType();
+
+        if (kind == MessageType.ScheduleUpgrade) {
+            return scheduleUpgrade;
+        } else if (kind == MessageType.CancelUpgrade) {
+            return cancelUpgrade;
+        } else if (kind == MessageType.RecoverTokens) {
+            return recoverTokens;
+        } else if (kind == MessageType.RegisterAsset) {
+            return registerAsset;
+        } else if (kind == MessageType.Request) {
+            return request;
+        } else if (kind == MessageType.NotifyPool) {
+            return notifyPool;
+        } else if (kind == MessageType.NotifyShareClass) {
+            return notifyShareClass;
+        } else if (kind == MessageType.NotifyPricePoolPerShare) {
+            return notifyPricePoolPerShare;
+        } else if (kind == MessageType.NotifyPricePoolPerAsset) {
+            return notifyPricePoolPerAsset;
+        } else if (kind == MessageType.NotifyShareMetadata) {
+            return notifyShareMetadata;
+        } else if (kind == MessageType.UpdateShareHook) {
+            return notifyShareHook;
+        } else if (kind == MessageType.InitiateTransferShares) {
+            return initiateTransferShares;
+        } else if (kind == MessageType.ExecuteTransferShares) {
+            return executeTransferShares;
+        } else if (kind == MessageType.UpdateRestriction) {
+            return updateRestriction;
+        } else if (kind == MessageType.UpdateContract) {
+            return updateContract;
+        } else if (kind == MessageType.RequestCallback) {
+            return requestCallback;
+        } else if (kind == MessageType.UpdateVault) {
+            return updateVault;
+        } else if (kind == MessageType.SetRequestManager) {
+            return setRequestManager;
+        } else if (kind == MessageType.UpdateBalanceSheetManager) {
+            return updateBalanceSheetManager;
+        } else if (kind == MessageType.UpdateHoldingAmount) {
+            return updateHoldingAmount;
+        } else if (kind == MessageType.UpdateShares) {
+            return updateShares;
+        } else if (kind == MessageType.MaxAssetPriceAge) {
+            return maxAssetPriceAge;
+        } else if (kind == MessageType.MaxSharePriceAge) {
+            return maxSharePriceAge;
+        } else {
+            return 0; // Unreachable
```

Comment:
Maybe we return `type(uint128).max` here, as a way to error out (won't be possible to be paid)

Reply:
Makes sense 👍🏻